src/components/UI/Dropdown/index.js

```diff
@@ -1,19 +1,101 @@
-import React from "react";
-import i18next from "i18next";
-// Styles
-import { Container } from "./dropdown.styled";
-
-export function Dropdown() {
-  const handleLanguageChange = (lang = "en") => {
-    i18next.changeLanguage("sr");
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { IoIosArrowDown } from 'react-icons/io';
+
+export function Dropdown(props) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [handleOption, setHandleOption] = useState(props?.list[0] || []);
+
+  const handleChange = () => {
+    if (props?.list.length == 0) return;
+    setIsOpen(!isOpen);
+  };
+
+  const handleClicked = (value) => {
+    setHandleOption(value);
+    setIsOpen(false);
+    props.onSelect(value);
   };
 
   return (
-    <Container>
-      <select onChange={handleLanguageChange}>
-        <option>en</option>
-        <option>sr</option>
-      </select>
-    </Container>
+    <Main>
+      <DropDownContainer>
+        <DropDownHeader onClick={handleChange} fontSize={props.fontSize}>
+          {handleOption.name}
+          {props?.list.length == 0 && (
+            <Information>
+              <p>Loading...</p>
+            </Information>
+          )}
+          {props?.list.length !== 0 && <IoIosArrowDown />}
+        </DropDownHeader>
+
+        {isOpen && (
+          <DropDownListContainer>
+            {props?.list.length > 0 ? (
+              <DropDownList>
+                {props?.list.map((item) => (
+                  <ListItem onClick={() => handleClicked(item)}>
+                    {item.name}
+                  </ListItem>
+                ))}
+              </DropDownList>
+            ) : (
+              'Loading...'
+            )}
+          </DropDownListContainer>
+        )}
+      </DropDownContainer>
+    </Main>
   );
 }
+
+const Main = styled('div')`
+  font-family: sans-serif;
+`;
+
+const DropDownContainer = styled('div')``;
+
+const DropDownHeader = styled('div')`
+  display: flex;
+  padding: 30px;
+  font-weight: 700;
+  font-size: ${(props) => props.fontSize || ''};
```

Comment:
as mentioned in a previous comment, should be done by styling this element instead of passing through props

Reply:
i understand everything. It will be done.